added_values.go

```diff
@@ -158,7 +158,7 @@ func (i valueNumber) Equals(other Value) bool {
 		return i.ToInt() == o.ToInt()
 	}
 	if o, ok := other.(*Object); ok {
-		return i.Equals(o.self.toPrimitiveNumber())
+		return i.Equals(o.toPrimitiveNumber())
```

Comment:
Needed this after cherry picking this [commit](https://github.com/mongodb-forks/goja/commit/28ee0ee714f3e8218c14e2cd2d7a67ec94848c21)... there are several other changes like this.

Reply:
This was due to objectImpl (the type of `self`) not having a field or method toPrimitiveNumber.